src/styles/main.scss

```diff
@@ -0,0 +1,87 @@
+$day-size: 100px;
+$day-border: 1px;
+$day-background: #eee;
+$day-hover-color: #ffbfcb;
+$day-font: Arial, sans-serif;
+$day-font-size: 30px;
+$columns: 7;
+$gap: 1px;
+$padding: 10px;
+$transition-duration: 0.5s;
+
+body {
+  margin: 0;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  overflow: hidden;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: flex-start;
+  gap: $gap;
+  margin: auto;
+  max-width: 706px;
+  padding: $padding;
+
+  &__day {
+    position: relative;
+    box-sizing: border-box;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    height: $day-size;
+    width: $day-size;
+    border: $day-border solid black;
+    background-color: $day-background;
+    font-family: $day-font;
+    font-size: $day-font-size;
+    cursor: pointer;
+    transition:
+      transform $transition-duration,
+      background-color $transition-duration;
+
+    &::before {
+      content: counter(day);
+      counter-increment: day;
```

Comment:
The use of CSS counters here is intended to number the days, but the counter-reset should be applied to the parent element of the days, not within the `&__day` itself. Consider moving `counter-reset: day;` to the `.calendar` class to ensure it resets correctly for each month.